contracts/ScheduledPaymentModule.sol

```diff
@@ -424,14 +476,58 @@ contract ScheduledPaymentModule is Module {
         );
         bool spTxStatus = exec(token, 0, spTxData, Enum.Operation.Call);
 
-        // execTransactionFromModule for gas reimbursement
+        // execTransactionFromModule for percentage fee
+        bytes memory feeTxData = abi.encodeWithSelector(
+            0xa9059cbb,
+            IConfig(config).getFeeReceiver(),
+            calculatePercentageFee(amount, fee)
+        );
+        bool feeTxStatus = exec(token, 0, feeTxData, Enum.Operation.Call);
+
+        // execTransactionFromModule for fixed fee and gas reimbursement
         bytes memory gasTxData = abi.encodeWithSelector(
             0xa9059cbb,
             IConfig(config).getFeeReceiver(),
-            executionGas.mul(gasPrice)
+            executionGas.mul(gasPrice).add(calculateFixedFee(gasToken, fee))
         );
         bool gasTxStatus = exec(gasToken, 0, gasTxData, Enum.Operation.Call);
 
-        return spTxStatus && gasTxStatus;
+        return spTxStatus && feeTxStatus && gasTxStatus;
+    }
+
+    function calculateFixedFee(address token, Fee calldata fee)
+        private
+        returns (uint256)
+    {
+        (uint256 usdRate, uint8 decimals) = IExchange(exchange).exchangeRateOf(
+            token
+        );
+        require(
+            decimals == feeFixedUsdDecimals(),
+            "unexpected decimals value for token price"
+        );
+
+        require(usdRate > 0, "exchange rate cannot be 0");
+        uint8 tokenDecimals = ERC20(token).decimals();
+        uint256 ten = 10;
+        return fee.fixedUSD.mul(ten**tokenDecimals).div(usdRate);
```

Comment:
There is a lib for using decimals in solidity that I've been meaning to try out for awhile. I'd suggest giving that a shot. We used it back in this project: https://github.com/cardstack/card-commerce-protocol/blob/master/contracts/Decimal.sol (I'm not sure of the original source for this).

Reply:
I've used D256 for the fee data type